### PR TITLE
[Compose] - 2656 - Do not rely on channel members to display message list header subtitle

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2656

### 🎯 Goal

We rely on the `Channel.members` field to show the number of members in a channel. Also, we use it to count and display the number online users. But the `Channel.members` is unreliable as it only contains the first 30 members of a channel, that's why we need to find another way.

### 🛠 Implementation details

We need to use `Channel.member_count` and `Channel.watcher_count` fields to display the `MessageListHeader` subtitle.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channel/info/ChannelInfo.kt [hunk 1]
```diff
@@ -78,8 +79,11 @@ public fun ChannelInfo(
         ?.role
         ?.let { it == "admin" || it == "owner" }
         ?: false
-
-    val otherMembers = channelMembers.filter { it.user.id != currentUser?.id }
+    val membersToDisplay = if (selectedChannel.isOneToOne(currentUser)) {
+        channelMembers.filter { it.user.id != currentUser?.id }
+    } else {
+        channelMembers
+    }
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channel/info/ChannelInfo.kt:86]: Displaying the current user in the list of channel members like other SDKs. Also, makes sense as we, for example, have a caption that the channel contains 3 members, and we actually have 3 use avatars below (including the current user).

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelUtils.kt [hunk 2]
```diff
@@ -84,21 +84,26 @@ public fun Channel.isOneToOne(currentUser: User?): Boolean {
...
-                count,
-                count,
-                otherMembers.count { it.user.online }
+            val memberCountString = context.resources.getQuantityString(
+                R.plurals.stream_compose_member_count,
+                memberCount,
+                memberCount
             )
+
+            return if (watcherCount > 0) {
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelUtils.kt:98]: Discussed with the backend team, and they claim that we can rely on the `watcherCount` field, which actually represents online users (not only the users who are watching the channel, but also connected to the WS). We might have some inconsistencies for some time as we had permissions disabled in the dashboard.

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channel/info/ChannelInfo.kt (hunk 1 above)
Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channel/info/ChannelInfo.kt:86]: Makes total sense, nicely done!

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelUtils.kt (hunk 2 above)
Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ChannelUtils.kt:98]: I've seen the discussion - I think if they stand behind it, then that's completely okay. Especially if we can show the correct number of people that are in the channel and not just the ones we paginated for

**Review by filbabic — APPROVED** [review]
Looks good!

**Pull request merged**